knimin/handlers/barcode_util.py

```diff
@@ -172,8 +202,61 @@ def _build_email(self, login_user, barcode, email_type,
         return subject, body_message
 
 
+def align_with_qiita_categories(samples, categories):
+    # obviously should instead align to the actual sample metadata...
+    aligned = {}
+    for s in samples:
+        aligned[s] = {c: 'LabControl test' for c in categories}
+    return aligned
+
+
+@set_access(['Scan Barcodes'])
+class PushQiitaHandler(BaseHandler):
+    executor = concurrent.futures.ThreadPoolExecutor(5)
+    study_id = config.qiita_study_id
+    qclient = get_qiita_client()
+
+    @concurrent.run_on_executor
+    def _push_to_qiita(self, study_id, samples):
+        # TODO: add a mutex or block to ensure a single call process at a time
```

Comment:
Should we have this as an issue?

Reply:
if you want to add. TODOs are typical in some of our codebases too. The likelihood of multiple executions here is low, and even then, sending the same sample multiple times to qiita is allowed as it's treated like an update

Reply:
Note that depending on the scenario an update will take hours vs. minutes as Qiita, currently, will need to update the sample in all the preps where is present. Now, in my experiences, TODOs in the code base never get revisited cause they are not easy to find; while issues do; thus, I try to open an issue for each of the TODOs. BTW there is a cool feature in github that allows you to create an issue from any of these conversations. New issue: https://github.com/biocore/labadmin/issues/253